processor/src/decoder/mod.rs

```diff
@@ -235,6 +236,7 @@ pub struct Decoder {
     aux_hints: AuxTraceHints,
     in_debug_mode: bool,
     operations: Vec<Operation>,
+    decorators: DecoratorList,
```

Comment:
Do we need to track all decorators here? Maybe this can be just a vector of `AsmOpInfo`? This may also make building of VM state a bit simpler.

Reply:
Changed `decorators: DecoratorList` to `asmop_map: BTreeMap<usize, AsmOpInfo>`

> Regarding 4.4% - I think that's ok for execute_iter?. Do you know if regular execute() is affected?
No change for `execute()`.

Reply:
Changed `decorators: DecoratorList` to `asmop_list: Vec<(usize, AsmOpInfo)>` in `Decoder` and changed decorators: `Option<Vec<Decorator>>` to `asmop: Option<AsmOpInfo>` in `VmState`.

    Regarding 4.4% - I think that's ok for execute_iter?. Do you know if regular execute() is affected?

No change for execute().